facebook-commerce.php

```diff
@@ -914,7 +934,23 @@ function on_product_save( $wp_id ) {
 			$sync_mode = Admin::SYNC_MODE_SYNC_AND_HIDE;
 		}
 
-		if ( ! $product->is_type( 'variable' ) ) {
+		$products_to_delete_from_facebook = $this->get_removed_from_sync_products_to_delete();
+
+		if ( $product->is_type( 'variable' ) ) {
+
+			// check variations for deletion
+			foreach ( $products_to_delete_from_facebook as $delete_product_id ) {
+
+				$delete_product = wc_get_product( $delete_product_id );
+
+				if ( Products::is_sync_enabled_for_product( $delete_product ) ) {
```

Comment:
We should check that that `wc_get_product()` returned a product object before passing `$delete_product` to `Products::is_sync_enabled_for_product()`.